sdk/computefleet/arm-computefleet/package.json

```diff
@@ -0,0 +1,211 @@
+{
+  "name": "@azure/arm-computefleet",
+  "version": "1.0.0-beta.1",
+  "description": "Azure Fleet Service",
+  "engines": {
+    "node": ">=18.0.0"
+  },
+  "sideEffects": false,
+  "autoPublish": false,
+  "tshy": {
+    "exports": {
+      "./package.json": "./package.json",
+      ".": "./src/index.ts",
+      "./api": "./src/api/index.ts",
+      "./models": "./src/models/index.ts",
+      "./api/operations": "./src/api/operations/index.ts",
+      "./api/fleets": "./src/api/fleets/index.ts"
+    },
+    "dialects": [
+      "esm",
+      "commonjs"
+    ],
+    "esmDialects": [
+      "browser",
+      "react-native"
+    ],
+    "selfLink": false
+  },
+  "type": "module",
+  "keywords": [
+    "node",
+    "azure",
+    "cloud",
+    "typescript",
+    "browser",
+    "isomorphic"
+  ],
+  "author": "Microsoft Corporation",
+  "license": "MIT",
+  "files": [
+    "dist",
+    "README.md",
+    "LICENSE",
+    "review/*"
+  ],
+  "sdk-type": "client",
+  "repository": "github:Azure/azure-sdk-for-js",
+  "bugs": {
+    "url": "https://github.com/Azure/azure-sdk-for-js/issues"
+  },
+  "prettier": "@azure/eslint-plugin-azure-sdk/prettier.json",
+  "//metadata": {
+    "constantPaths": [
+      {
+        "path": "src/rest/azureFleetClient.ts",
+        "prefix": "package-version"
+      }
+    ]
+  },
+  "dependencies": {
+    "@azure-rest/core-client": "^2.1.0",
+    "@azure/core-auth": "^1.6.0",
+    "@azure/core-rest-pipeline": "^1.5.0",
+    "@azure/logger": "^1.0.0",
+    "tslib": "^2.6.2",
+    "@azure/core-lro": "3.0.0",
+    "@azure/abort-controller": "^2.1.2",
+    "@azure/core-paging": "^1.5.0"
+  },
+  "devDependencies": {
+    "dotenv": "^16.0.0",
+    "@microsoft/api-extractor": "^7.40.3",
+    "@types/node": "^18.0.0",
+    "eslint": "^8.55.0",
+    "prettier": "^3.2.5",
+    "rimraf": "^5.0.5",
+    "mkdirp": "^3.0.1",
+    "typescript": "~5.4.5",
+    "tshy": "^1.11.1",
```

Comment:
We should update to 2.x for vitest.

Reply:
Just curious if this upgrade is trying to resolve this browser test failure issue https://github.com/Azure/azure-sdk-for-js/issues/30297? And I notice tshy has released 3.x several days right after 2.x, should we just use 3.x 
https://www.npmjs.com/package/tshy?activeTab=versions

Reply:
I have tried to update tshy to 2.x and re-run the case. but I got the same issue,
Then I tried to update thsy to 2.x or 3.x and update vitest and it's related package vitest/browser and vitest/coverage-istanbul to 2.x, I got following error message:
```
test/public/mongoclusterOperations.spec.ts(35,37): error TS2345: Argument of type 'TaskContext<Test<{}> | Custom<{}>> & TestContext' is not assignable to parameter of type 'VitestTestContext'.
  Types of property 'task' are incompatible.
    Type 'Readonly<Test<{}> | Custom<{}>>' is not assignable to type 'VitestTask'.
      Type 'Readonly<Test<{}>>' is not assignable to type 'VitestTask'.
        Types of property 'suite' are incompatible.
          Type 'Suite | undefined' is not assignable to type 'VitestSuite'.
            Type 'undefined' is not assignable to type 'VitestSuite'.
```

code :
```
export async function createRecorder(context: VitestTestContext): Promise<Recorder> {
  const recorder = new Recorder(context);
  await recorder.start(recorderEnvSetup);
  return recorder;
}

beforeEach(async (context) => {
    process.env.SystemRoot = process.env.SystemRoot || "C:\\Windows";
    recorder = await createRecorder(context);
    ...
```